### PR TITLE
Sensors simulation mode

### PR DESCRIPTION
**Pull request by Reznic**
New command line flag added: --simulate (-s)
which causes all hardware drivers to be initialized with mocks, instead of the real drivers.
If run on a laptop, we allways enter simulation mode.
If run on raspberry-pi, the default is real sensors mode. and can be changed with the flag.

### CODE REVIEW

File: drivers/mocks/mock_wd_driver.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import time
+
+
+class MockWdDriver(object):
+    WD_SIGNAL_LEN = 0.05
```
Comment (avrahamshukron) [inline drivers/mocks/mock_wd_driver.py]: 50ms for WD refresh?! that is **a lot**

File: drivers/wd_driver.py [hunk 2]
```diff
@@ -1,26 +1,18 @@
...
-import os
 
-OS_RASPB = os.uname()[1] == 'raspberrypi'
-if OS_RASPB:
-    import RPi.GPIO as GPIO
+import RPi.GPIO as GPIO
 
 
 class WdDriver(object):
     WD_SIGNAL_LEN = 0.05
```
Comment (avrahamshukron) [inline drivers/wd_driver.py]: 50ms for WD refresh?! that is **a lot**

File: main.py [hunk 3]
```diff
@@ -31,11 +30,13 @@ def configure_logging(level, store):
     logger.addHandler(fh)
     logger.addHandler(ch)
     logger.disabled = not store.log_enabled
+    return logger
 
 
 def parse_args():
     parser = argparse.ArgumentParser()
     parser.add_argument("--verbose", "-v", action="count", default=0)
+    parser.add_argument("--simulate", "-s", action='store_false')
```
Comment (avrahamshukron) [inline main.py]: Why `store_false`? if I pass `--simulate` then `simulate` should be `True`...

File: main.py [hunk 4]
```diff
@@ -51,13 +52,32 @@ def main():
...
+        pressure_sensor = MockHcePressureSensor()
+        watchdog = MockWdDriver()
+
+    else:
+        from drivers.hce_pressure_sensor import HcePressureSensor
+        from drivers.sfm3200_flow_sensor import Sfm3200
+        from drivers.wd_driver import WdDriver
+        flow_sensor = Sfm3200()
+        pressure_sensor = HcePressureSensor()
+        watchdog = WdDriver()
```
Comment (avrahamshukron) [inline main.py]: Too much code. There should be a factory method for each hardware component that takes a boolean parameter `simulate` and returns the appropriate driver:

```py
def create_wd_driver(simulate):
    pass  # do you magic here.

def create_air_flow_driver(simulate):
    pass  # do you magic here.

def create_air_pressure_driver(simulate):
    pass  # do you magic here.
```

**Comment by avrahamshukron** [comment]
I think maybe the WD driver should run in its own thread.
It will wait on a sync object, and when invoked it will kick the dog.

The UI thread will satisfy that sync object after each iteration.

This way the UI thread won't block on WD refresh

File: drivers/mocks/mock_wd_driver.py (hunk 1 above)
Comment (Reznic) [reply drivers/mocks/mock_wd_driver.py]: Ok that's another branch - the WD branch
Talk with Bruchim.
This mock method does the same sleep done in the real WD driver, writen by bruchim

File: main.py (hunk 3 above)
Comment (Reznic) [reply main.py]: Oops you're right

File: drivers/wd_driver.py (hunk 2 above)
Comment (JonBruchim) [reply drivers/wd_driver.py]: that's correct, when I will know the right parameters I'll set them.
could make it 5ms for now.
But it was set as 50 so i could validate the WD with a scope (validation was correct)

**Comment by zazohar** [comment]
> I think maybe the WD driver should run in its own thread.
> It will wait on a sync object, and when invoked it will kick the dog.
> 
> The UI thread will satisfy that sync object after each iteration.
> 
> This way the UI thread won't block on WD refresh

We must assure that ONLY the IU thread notify the WD, otherwise we may not invoke the buzzer in all error cases

File: main.py (hunk 4 above)
Comment (Reznic) [reply main.py]: Done

**Pull request merged**